File: tests/fixtures/sample.cs
using System;
using System.Collections.Generic;

interface IGreetable {
    string Greet(string name);
}

class Greeter : IGreetable {
    public string Greet(string name) {
        return $"Hello, {name}!";
    }
}

class Program {
    static Greeter CreateGreeter() {
        return new Greeter();
    }

    static void Main(string[] args) {
        Greeter g = CreateGreeter();
        string msg = g.Greet("World");
        Console.WriteLine(msg);
    }

    const string DefaultName = "World";
}

File: tests/fixtures/sample.go
package main

import "fmt"

type Greeter struct {
	Name string
}

func (g *Greeter) Greet() string {
	return fmt.Sprintf("Hello, %s!", g.Name)
}

func NewGreeter(name string) *Greeter {
	return &Greeter{Name: name}
}

func main() {
	g := NewGreeter("World")
	msg := g.Greet()
	fmt.Println(msg)
}

var DefaultName = "World"

// Interface
type Reader interface {
	Read(p []byte) (n int, err error)
}
